### DISCUSSION TITLE
Rename Pymem class constructor values to be more clear

### DISCUSSION THREAD

**Discussion by ughstudios** [srounet/Pymem#97] [General]
def __init__(self, process_name=None):

Can we rename process_name to "process"? And then add type hints like so:
```
    def __init__(self, process: str | int):
```

**Comment by StarrFox** [comment]
this is definitely something that would be nice to change; however I feel it would require a major version release since the argument isn't positional only i.e. some people may be doing `Pymem(process_name="abc.def")`

can see this original discussion in https://github.com/srounet/Pymem/pull/94
